lib/topological_inventory/mock_source/collector.rb

```diff
@@ -179,47 +143,13 @@ def targeted_refresh(notices)
         save_inventory(parser.collections.values)
       end
 
-      def save_inventory(collections, refresh_state_uuid = nil, refresh_state_part_uuid = nil)
-        return if collections.empty?
-
-        ingress_api_client.save_inventory(
-          :inventory => TopologicalInventoryIngressApiClient::Inventory.new(
-            :name                    => "OCP",
-            :schema                  => TopologicalInventoryIngressApiClient::Schema.new(:name => "Default"),
-            :source                  => source,
-            :collections             => collections,
-            :refresh_state_uuid      => refresh_state_uuid,
-            :refresh_state_part_uuid => refresh_state_part_uuid,
-          )
-        )
-      end
-
-      def sweep_inventory(refresh_state_uuid, total_parts, sweep_scope)
-        ingress_api_client.save_inventory(
-          :inventory => TopologicalInventoryIngressApiClient::Inventory.new(
-            :name               => "OCP",
-            :schema             => TopologicalInventoryIngressApiClient::Schema.new(:name => "Default"),
-            :source             => source,
-            :collections        => [],
-            :refresh_state_uuid => refresh_state_uuid,
-            :total_parts        => total_parts,
-            :sweep_scope        => sweep_scope,
-          )
-        )
-      end
-
-      # Should be overriden by subclass
-      def entity_types
-        []
-      end
-
-      def ingress_api_client
-        TopologicalInventoryIngressApiClient::DefaultApi.new
-      end
-
       def parser_class
         TopologicalInventory::MockSource::Parser
       end
+
+      def connection_for_entity_type(_entity_type)
+        :unused
```

Comment:
is this used somewhere? :-) maybe `nil` will be enough ?

Reply:
the original method using call to `connection_for_entity_type` (taken from openshift collector) ends whole thread when nil, so that's the reason I used something else now, to be compatible. 
I agree it looks weird, I'll take a look at it